sw/host/opentitanlib/src/bootstrap/eeprom.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+
+use crate::app::TransportWrapper;
+use crate::bootstrap::{Bootstrap, UpdateProtocol};
+use crate::spiflash::SpiFlash;
+use crate::transport::Capability;
+
+/// Implements the SPI EEPROM bootstrap protocol.
+pub struct Eeprom;
+
+impl Eeprom {
+    /// Creates a new `Eeprom` protocol updater.
+    pub fn new() -> Self {
+        Eeprom
+    }
+}
+
+impl UpdateProtocol for Eeprom {
+    fn verify_capabilities(
+        &self,
+        _container: &Bootstrap,
+        transport: &TransportWrapper,
+    ) -> Result<()> {
+        transport
+            .capabilities()?
+            .request(Capability::GPIO | Capability::SPI)
+            .ok()?;
+        Ok(())
+    }
+
+    fn uses_common_bootstrap_reset(&self) -> bool {
+        true
+    }
+
+    /// Performs the update protocol using the `transport` with the firmware `payload`.
+    fn update(
+        &self,
+        container: &Bootstrap,
+        transport: &TransportWrapper,
+        payload: &[u8],
+    ) -> Result<()> {
+        let spi = container.spi_params.create(transport)?;
+        let flash = SpiFlash::from_spi(&*spi)?;
+        flash.chip_erase(&*spi)?;
```

Comment:
We are initializing the flash with a reference to the spi. So why do we still need to pass the spi in every call to the flash methods?

Maybe this comment is out of scope for this PR.

Reply:
Just started looking at this part of opentitanlib. It looks like `SpiFlash` just holds the characteristics of the memory device and expects the actual IO to be used for communication passed in to its methods, i.e. dependency injection, but I'm not sure why it doesn't cache it at creation. @cfrantz might be able to clarify.